base/src/main/java/io/spine/query/package-info.java

```diff
@@ -19,7 +19,386 @@
+ *
+ * <p>In addition to the columns declared in the Entity state, the query API allows declaring
+ * the custom columns. They are intended to handle the cases in which some computed on-the-fly data
+ * should be stored along with the record. It may the time of entity creation, the role of the
+ * user created the record, whether the entity is deleted or archived etc. That is, something
+ * which isn't included into the definition of the original {@code Message} type of the record.
+ *
+ * <pre>
+ * // The column presumably defined as a custom column for querying.
+ * class ArchivedColumn extends CustomColumn<EntityWithLifecycle, Boolean> {
+ *
+ *   {@literal @Override }
+ *    public ColumnName name() {
+ *        return ColumnName.of("archived");
+ *    }
+ *
+ *   {@literal @Override }
+ *   {@literal public Class<Boolean>} type() {
+ *        return Boolean.class;
+ *    }
+ *
+ *   {@literal @Override }
+ *    public Boolean valueIn(EntityWithLifecycle source) {
+ *        return source.isArchived();
+ *    }
+ * }
+ * </pre>
+ *
+ * <p>Where {@code EntityWithLifecycle} is a sample interface serving as a source of the lifecycle
+ * values for the Entity types.
+ *
+ * <pre>
+ * // Custom columns that define Entity lifecycle.
+ * public enum Lifecycle {
+ *
+ *    ARCHIVED(new ArchivedColumn()),
+ *
+ *    DELETED(new DeletedColumn());
+ *
+ *   {@literal private final CustomColumn<?, Boolean>} column;
+ *
+ *   {@literal Lifecycle(CustomColumn<?, Boolean>} column) {
+ *        this.column = column;
+ *    }
+ *
+ *    // Returns the column declaration.
+ *   {@literal public CustomColumn<?, Boolean>} column() {
+ *        return column;
+ *    }
+ * }
+ * </pre>
+ *
+ * <p>Such an approach allows defining the singleton instances of the columns and using
+ * them in querying:
+ *
+ * <pre>
+ *     ProjectView.Query queryForDeleted =
+ *        ProjectView.newQuery()
+ *                   .where(DELETED.column(), true)
+ *                   .build();
+ * </pre>
+ *
+ * <h4>Ordering and limit</h4>
+ *
+ * <p>Similar to the plain record query API, it is possible to set the same query parameters
+ * when querying the Entity state records:
+ *
+ * <pre>
+ *     ProjectView.Query query = ProjectView
+ *        .newQuery()
+ *        .orderBy(daysSinceStarted(), ASC)
+ *        .orderBy(projectName(), ASC)
+ *        .orderBy(wasReassigned(), DESC)
+ *        .limit(10)
+ *        .build();
+ * </pre>
+ *
+ * <p>Here {@code daysSinceStarted()}, {@code projectName()} and {@code wasReassigned()} are three
+ * static methods generated in {@code ProjectView.Column} class. Each of them returns
+ * the declaration of the Entity column corresponding to the Message field with
+ * the {@code (column)} option.
+ *
+ * <h4>Field masks</h4>
+ *
+ * <p>It is also possible to set the mask for each resulting Protobuf message:
+ *
+ * <pre>
+ *     FieldMask mask = fieldMaskWith(status());
+ *     ProjectView.Query query = ProjectView.newQuery()
```

Comment:
Assuming we are going to migrate to Java 11 soon, what if we apply the `var` syntax to the code examples _already_? It would make code samples a bit more compact. Readers who are still under Java 8 would still understand them.

Reply:
@alexander-yevsyukov we'll then have to do that in every case, not only in this document.
I'd prefer to phase it out until we really migrate, rather than do that just in a few cases out of dozens.